data-prepper-expression/src/main/java/org/opensearch/dataprepper/expression/JoinExpressionFunction.java

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.dataprepper.expression;
+
+import org.opensearch.dataprepper.model.event.Event;
+
+import java.util.HashMap;
+import java.util.List;
+import javax.inject.Named;
+import java.util.Map;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+@Named
+public class JoinExpressionFunction implements  ExpressionFunction {
+
+    private static final String FUNCTION_NAME = "join";
+
+    @Override
+    public String getFunctionName() {
+        return FUNCTION_NAME;
+    }
+
+    @Override
+    public Object evaluate(final List<Object> args, Event event, Function<Object, Object> convertLiteralType) {
+        if (args.isEmpty() || args.size() > 2) {
+            throw new IllegalArgumentException(FUNCTION_NAME + "() takes one or two arguments");
+        }
+
+        final List<String> argStrings;
+        try {
+            argStrings = args.stream()
+                    .map(arg -> ((String)arg).trim())
+                    .collect(Collectors.toList());
+        } catch (Exception e) {
+            throw new IllegalArgumentException(
+                    "Arguments in " + FUNCTION_NAME + "() function should be of Json Pointer type or String type");
+        }
+
+        final String delimiter;
+        final String sourceKey;
+        if (argStrings.size() == 2) {
+            final String trimmedDelimiter = argStrings.get(0).substring(1, argStrings.get(0).length() - 1);
+
+            // remove slashes used to escape comma
+            delimiter = trimmedDelimiter.replace("\\\\,", ",");
+            sourceKey = argStrings.get(1);
+        } else {
+            delimiter = ",";
+            sourceKey = argStrings.get(0);
```

Comment:
Does this mean, we allow `join(/aaa, /bbb, /ccc, /ddd)` and it takes just "/aaa" and ignores rest? If yes, that seems confusing that we silently ignore other parameters.

Reply:
Currently it only supports one or two parameters. There's a check on number of arguments at the beginning, so `join(/aaa, /bbb, /ccc, /ddd)` won't pass that.